File: lib/backend/user_feedback.dart
import 'package:flutter/material.dart';

import '../utils.dart';

/// User feedback.
@immutable
class UserFeedback {
  final String text;

  // Creation datetime
  final DateTime datetime;
  final String? uid;

  const UserFeedback({required this.text, required this.datetime, this.uid});

  DynMap serialize() {
    return {
      'text': text,
      'datetime':
          datetime.toIso8601String(), // Convert the DateTime object to a string
      'uid': uid,
    };
  }
}

File: functions/test/user/data/user_feedback.ts
import { UserFeedback } from "../../../src/user/feedback";
/**
 * A dummy feedback.
 */
export const FEEDBACK_JSON_0 = {
  text: "test-feedback",
  datetime: new Date(),
  uid: "test-user",
};

export const FEEDBACK_0 = new UserFeedback(
  FEEDBACK_JSON_0.text,
  FEEDBACK_JSON_0.datetime,
  FEEDBACK_JSON_0.uid
);
